Feature user model

banjodayo39 opened the pull request:
## Description
Please include a summary of the change and relevant motivation and context. 
List any dependencies that are required for this change.

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #(issue number 1) #(issue number 2 if applicable)

## How Has This Been Tested?
Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code etc.

## Screenshots (if applicable, else remove this line / section)


## Checklist:
<!--- Put an `x` in all the boxes that apply ! -->
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added necessary inline code documentation
- [ ] I have added tests that prove my fix is effective and that this feature works
- [ ] New and existing unit tests pass locally with my changes

banjodayo39 committed:
added user profile

banjodayo39 committed:
added userprofile model

elizabethadegbaju commented on src/goodHealthApp/models.py:
```diff
@@ -1,16 +1,40 @@
-    UserProfile Model.
-    """
     user = models.OneToOneField(User, on_delete=models.CASCADE)
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    email = models.CharField(max_length=100)
+    password = models.CharField(max_length=100)
```
These should be removed. This model has a one-to-one field with the default django user model with these existing fields.

elizabethadegbaju commented on src/goodHealthApp/models.py:
```diff
@@ -1,16 +1,40 @@
+    phone_number = models.CharField(max_length=100)
+    avatar = models.ImageField(upload_to=get_upload_image_dir, null=True, blank=True)
+
+    def __str__(self):  # __unicode__ for Python 2
+        return f"{self.first_name} {self.last_name}"
```
This should be changed too with respect to the required changes in the userprofile model

elizabethadegbaju committed:
Delete 0005_auto_20200514_1121.py

The changes to the model reflected in this migration are not accepted. Rerun the makemigrations command after your edits to the userprofie model.

elizabethadegbaju commented on src/goodHealthApp/models.py:
```diff
@@ -1,16 +1,40 @@
+
+
+def get_upload_image_dir(instance, filename):
+    return "uploads/{mimeType}/{userid}/{file}".format(mimeType="images", userid=instance.user.username, file=filename)
```
I think the depth of this path is not needed considering each user can only have one avatar at any point in that. consider using a single folder for the avatars where each avatar bears the user's username as the filename. something like "uploads/avatars/elizabethadegbaju.png" rather than "uploads/images/elizabethadegbaju/filenameasuploaded.png"
